tests/unit/explore_test.sh

```diff
@@ -172,6 +185,47 @@ function test_explore_git()
   }
 }
 
+function test_explore_context()
+{
+  local -r current_path="$PWD"
+  local EXPECTED_CONTEXT='3'
+  local EXPECTED_MATCH='avoid'
+  local MSG_OUT='7'
+
+  cd "$SHUNIT_TMPDIR" || {
+    fail "(${LINENO}) It was not possible to move to temporary directory"
+    return
+  }
+
+  # Check the number of output lines
+  output=$(explore_main --show-context=3 'avoid' codestyle_error.c | wc --lines)
+  assertEquals "(${LINENO})" "$MSG_OUT" "$output"
+
+  output=$(explore_main --show-context="$EXPECTED_CONTEXT" "$EXPECTED_MATCH" codestyle_error.c)
+
+  # Check if the expected match and context lines are present in the output
+  assert_substring_match "Expected match not found!" ${LINENO} "${EXPECTED_MATCH}" "$output"
+
+  # Check context lines below the match
+  for ((i = 1; i <= EXPECTED_CONTEXT; i++)); do
+    CONTEXT_LINE=$((i + 4)) # Assuming match is on line 4
+    CONTEXT_LINE_CONTENT=$(echo "$output" | head -n $CONTEXT_LINE | tail -n 1)
```

Comment:
use `printf` instead of `echo`.
https://kworkflow.org/content/codingstyle.html#use-printf-instead-of-echo

Also use --lines= instead of -n

and add double quotes around CONTEXT_LINE.